src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlSecurityToken.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using CoreWCF.Security;
+using MSIdentitySAML = Microsoft.IdentityModel.Tokens.Saml;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    public class SamlSecurityToken : SecurityToken
+    {
+        protected SamlSecurityToken()
+        {
+        }
+
+        internal SamlSecurityToken(MSIdentitySAML.SamlSecurityToken samlSecurityToken)
+        {
+            InternalSamlSecurityToken = samlSecurityToken;
+            Assertion = InternalSamlSecurityToken.Assertion;
+        }
```

Comment:
Should you be initializing SamlStatements here? Also null check on samlSecurityToken. As it's an internal method it shouldn't be null so use an Fx.Assert to remove the runtime cost but still catch bugs during testing.